src/commands/diff.rs

```diff
@@ -223,15 +223,20 @@ async fn diff_impl(
 
     // Get the name of the existing Pull Request branch, or constuct one if
     // there is none yet.
+
+    let title = message
+        .get(&MessageSection::Title)
+        .map(|t| &t[..])
+        .unwrap_or("");
+
     let pull_request_branch_name = match &pull_request {
         Some(pr) => get_branch_name_from_ref_name(&pr.head)?.to_string(),
-        None => config.get_new_branch_name(
-            &git.get_all_ref_names()?,
-            message
-                .get(&MessageSection::Title)
-                .map(|t| &t[..])
-                .unwrap_or(""),
-        ),
+        None => config.get_new_branch_name(&git.get_all_ref_names()?, &title),
+    };
+
+    let base_branch_name = match &pull_request {
+        Some(pr) => get_branch_name_from_ref_name(&pr.base)?.to_string(),
```

Comment:
I think we have a problem here when updating an existing pull request that is currently based on `master`. In that case `base_branch_name` will be `master`.

The commit may have been rebased on another off-master commit in the meantime, which means we are creating a base branch now. We can't call it `master` though.

Reply:
I think I've fixed this by basing `base_branch_name` off `base_branch` rather than `pull_request`, which already handles checking if there's a non-master base branch.